service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -0,0 +1,781 @@
+#!/bin/bash
+#
+# Name: startup.sh
+#
+# Description
+#   Initialization script to setup Terra configurations in a Dataproc cluster manager node
+#
+# Execution details
+#   This script is executed as root on all Dataproc nodes.
+#
+#   Note that the guest attribute is set to DONE whether the script runs successfully or not.
+#
+# How to test changes to this file:
+# Currently, the only way is to use swagger to create a new cluster with the startup script gs url passed into the 'startup-script-uri' metadata field.
+#   TODO: Pending CLI support to create a cluster with a custom startup script via CLI
+#
+# Integration Tests
+#   Please also make sure integration test `PrivateControlledDataprocClusterStartup` passes. Refer to
+#   https://github.com/DataBiosphere/terra-workspace-manager/tree/main/integration#Run-nightly-only-test-suite-locally
+#   for instruction on how to run the test.
+
+# Only run on the dataproc manager node. Exit silently if otherwise.
+readonly ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+if [[ "${ROLE}" != 'Master' ]]; then exit 0; fi
+
+# Only run on first startup
+if [[ -f /etc/startup_was_launched ]]; then exit 0; fi
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+# The linux user that JupyterLab will be running as. It's important to do some parts of setup in the
+# user space, such as setting Terra CLI settings which are persisted in the user's $HOME.
+# This post startup script is not run by the same user.
+readonly DATAPROC_USER="dataproc"
```

Comment:
So by default, the dataproc jupyter optional component configured the jupyter service to run jupyter as root. I changed it to the dataproc user and gave it sudo (line 175).